storage/remote/client.go

```diff
@@ -267,28 +267,38 @@ func (c *Client) Store(ctx context.Context, req []byte, attempt int) error {
 	if err != nil {
 		// Errors from Client.Do are from (for example) network errors, so are
 		// recoverable.
-		return RecoverableError{err, defaultBackoff}
+		return WriteResponseStats{}, RecoverableError{err, defaultBackoff}
 	}
 	defer func() {
 		io.Copy(io.Discard, httpResp.Body)
 		httpResp.Body.Close()
 	}()
 
+	rs, err := ParseWriteResponseStats(httpResp)
+	if err != nil {
+		// There were some response header values we can't parse, so fail fast here.
+		return WriteResponseStats{}, err
```

Comment:
Correct. I decided to not trust any of the headers if once header was malformed - otherwise we might have a skewed observation 🤔 But maybe we would fail request we could be otherwise valid? Not sure.

Reply:
I feel like using the results if any were present is valid. If it isn't, we might as well modify the `ParseWriteResponseStats` to fail fast and return the first error instead of attempting to parse all headers?

Reply:
fair, fixing